connection.go

```diff
@@ -63,3 +63,29 @@ type session struct {
 func (s session) Connection() Connection {
 	return s.connection
 }
+
+// Tx embeds dbr.Tx and norm Session
+type Tx interface {
+	Session
+	Commit() error
+	Rollback() error
+	RollbackUnlessCommitted()
+}
+
+// tx implements Tx interface
+type tx struct {
+	*dbr.Tx
+	connection Connection
+}
+
+// Connection returns norm Connection
+func (t tx) Connection() Connection {
+	return t.connection
+}
+
+// Begin returns a norm Tx which has wrapped a dbr.Tx
+// A real database connection has been aquired and is held by the enclosed sql.Tx instance
+func Begin(s Session) (Tx, error) {
+	dbrTx, err := s.Begin()
+	return &tx{dbrTx, s.Connection()}, err
+}
```

Comment:
I'm not a fan of adding this Method here, but its purpose is to return our Tx implementation. The other Begin method returns a dbr.Tx which is incompatible with where we pass norm Session.  I wish there was a better name of location for it.